examples/angular2/services/store.ts

```diff
@@ -0,0 +1,76 @@
+export class Todo {
+	completed: Boolean;
+	editing: Boolean;
+	title: String;
+	uid: String;
+	cleanTitle(title:String) {
+		return title.trim();
+	}
+	setTitle(title:String) {
+		this.title = this.cleanTitle(title);
+	}
+	constructor(title:String) {
+		this.uid = uuid.v4();
+		this.completed = false;
+		this.editing = false;
+		this.title = this.cleanTitle(title);
+	}
+}
+
+export class TodoStore {
+	todos: Array<Todo>;
+	constructor() {
+		let persistedTodos = store.get('angular2-todos') || [];
+		// Normalize back into classes
+		this.todos = persistedTodos.map( todo => {
+			let ret = new Todo(todo.title);
+			ret.completed = todo.completed;
+			ret.uid = todo.uid;
+			return ret;
+		});
+	}
+	_updateStore() {
+		store.set('angular2-todos', this.todos);
+	}
+	get(state:Object) {
+		return this.todos.filter((todo:Todo) => todo.completed == state.completed);
+	}
+	allCompleted() {
+		return this.todos.length == this.getCompleted().length;
+	}
+	setAllTo(toggler) {
+		this.todos.forEach((t:Todo) => t.completed = toggler.checked);
+		this._updateStore();
+	}
+	removeCompleted() {
+		this.todos = this.get({completed: false});
+	}
+	getRemaining() {
+		return this.get({completed: false});
+	}
+	getCompleted() {
+		return this.get({completed: true});
+	}
+	toggleCompletion(uid:String) {
+		this.todos.some((todo) => {
+			if (todo.uid === uid) {
+				todo.completed = !todo.completed;
+				return true;
+			}
+		});
+		this._updateStore();
+	}
+	remove(uid:String) {
+		this.todos.some((todo, i) => {
```

Comment:
> I'm assuming it's to short-circuit the loop.

Correct :sparkles: 

> You could use ES6 Map instead to avoid iterating on when you have to remove an item.

Oh cool let me look into that